feat(error-codes): Add codes to thrown errors

elliotttf opened the pull request:
This adds codes to thrown errors so that modules that depend on emdaer
may programatically react to thrown errors without parsing messages.

This relates to #4.

elliotttf committed:
feat(error-codes): Add codes to thrown errors

This adds codes to thrown errors so that modules that depend on emdaer
may programatically react to thrown errors without parsing messages.

flipactual commented:
Let's get a test in to make sure this doesn't get reverted at some point, then lgtm

elliotttf committed:
test(error-codes): Add tests, simplify constructor

This allows the error code to be set from the constructor rather
than as an additional step.

elliotttf committed:
docs(EmdaerError): Add jsdoc to the class definition
